KarhooUISDKTests/TestCases/Routing/AdyenCardRegistrationFlowSpec.swift

```diff
@@ -36,11 +39,38 @@ final class AdyenCardRegistrationFlowSpec: XCTestCase {
         testObject.setBaseView(mockBaseViewController)
     }
 
+    func testStartingDropIn() {
+        
+    }
+    
     private func cardRegistrationFlowCompletion(result: OperationResult<CardFlowResult>) {
-        self.capturedCardFlowResult = result
+        switch result {
+        case .completed(let result): cardRegistrationFlowCompletionResult = result
+        case .cancelledByUser: cardRegistrationCancelledByUser = true
+        }
     }
 
-    func testStartingDropIn() {
+    /**
+      * Given: The flow starts
+      * When: Currency and Amont is provided
+      * Then: AdyenPaymentMethodsRequest is made with the correct payload
+      */
+    func testFlowStart() {
+        simulateStartFlow()
+
+        XCTAssert(mockPaymentService.adyenPaymentMethodsCalled)
+        XCTAssertEqual(amount, mockPaymentService.adyenPaymentMethodsRequest?.amount?.value)
+        XCTAssertEqual(mockCardCurrencyCode, mockPaymentService.adyenPaymentMethodsRequest?.amount?.currency)
```

Comment:
Should we assert the supplier id on the request?

Reply:
The supplier ID is populated as part of the AdyenPaymentsRequest.

Reply:
the amount and currency are part of `adyenPaymentMethods` and supplierPartnerId is part of `adyenPayments` ... the first one is made as soon as the flow starts ... the other one is made later during the process